src/screens/RecoverPin.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import {
+  SafeAreaView,
+  StyleSheet,
+  Text,
+  View,
+} from 'react-native';
+import hathorLib from '@hathor/wallet-lib';
+import { connect } from 'react-redux';
+import { t } from 'ttag';
+import {
+  setRecoveringPin,
+  lockScreen,
+  setTempPin,
+  setLoadHistoryStatus,
+} from '../actions';
+import { guessPin } from '../utils';
+import SimpleButton from '../components/SimpleButton';
+import Spinner from '../components/Spinner';
+
+const mapStateToProps = (state) => ({
+  tempPin: state.tempPin,
+});
+
+const mapDispatchToProps = (dispatch) => ({
+  setRecoveringPin: (state) => dispatch(setRecoveringPin(state)),
+  lockScreen: () => dispatch(lockScreen()),
+  setTempPin: () => dispatch(setTempPin(null)),
+  setLoadHistoryStatus: (active, error) => dispatch(setLoadHistoryStatus(active, error)),
+});
+
+class RecoverPinScreen extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      erroed: false,
+      progress: 0,
+    };
+  }
+
+  componentDidMount() {
+    setTimeout(() => this.recoverPin(), 0);
+  }
+
+  goToReset() {
+    // navigate to reset screen
+    this.props.navigation.navigate('ResetWallet', {
+      hideBackButton: true,
+    });
+
+    // make sure we won't show loadHistory screen
+    this.props.setLoadHistoryStatus(false, false);
+  }
+
+  async recoverPin() {
+    const accessData = hathorLib.wallet.getWalletAccessData();
+    const [success, pin] = await guessPin(accessData, (progress) => {
+      this.setState({ progress });
+    });
+
+    if (!success) {
+      this.setState({
+        erroed: true,
+      });
+      return;
+    }
+
+    const currentPin = this.props.tempPin;
+    // update the encrypted words with the pin the user knows about
+    hathorLib.wallet.changePassword(pin, currentPin);
+
+    this.props.setRecoveringPin(false);
```

Comment:
Shouldn't we clean up the temporary PIN here?

Reply:
We should! Done in 291d59f80ff01a90d9d0c56ffe4caf4ce07b26f3

Reply:
Isn't it better to automatically do it when setting recovering pin to false?

Reply:
Great idea, fixed in 2eb4cc0d0b7d1ddae8960fb4b6a1b2ce6e95a47e

Reply:
Great! Thanks!